### ISSUE TITLE
Issues when changing dimensions during a teleport on fabric

### ISSUE THREAD

**Issue by foxii-cn** [WiIIiam278/HuskHomes#457]
In HuskHomes cross-server mode, whenever I die in The End and respawn at the spawn point, then use another command like `/server creative` to teleport to another sub-server, when I use `/back` to return, it triggers a bug. At this point, my client appears to have numerous error blocks, and there are also some strange creatures that can be moved.

![4@$Y_R%(PMWG2C~Q9RD%9VP](https://github.com/WiIIiam278/HuskHomes/assets/38043552/e4301503-0a19-4e84-b9dc-d7496b3753cc)

![{L_ 8U1M9IV0LB8Q VE}~ I](https://github.com/WiIIiam278/HuskHomes/assets/38043552/ae53001e-f496-492a-b3d5-e4bd85500847)

And when I quit the client and log in again, even bigger errors continue to occur.

```
[08:05:31] [Server thread/INFO]: Foxii lost connection: Disconnected
[08:05:31] [Server thread/INFO]: Foxii冲出去了
[08:05:31] [Server thread/ERROR]: Encountered an unexpected exception
java.lang.NullPointerException: Cannot invoke "it.unimi.dsi.fastutil.objects.ObjectSet.remove(Object)" because "$$4" is null
	at net.minecraft.class_3204.method_14051(class_3204.java:226) ~[server-intermediary.jar:?]
	at net.minecraft.class_3898.method_18714(class_3898.java:1073) ~[server-intermediary.jar:?]
	at net.minecraft.class_3898.method_18716(class_3898.java:1229) ~[server-intermediary.jar:?]
	at net.minecraft.class_3215.method_18753(class_3215.java:480) ~[server-intermediary.jar:?]
	at net.minecraft.class_3218$class_5526.method_31437(class_3218.java:1581) ~[server-intermediary.jar:?]
	at net.minecraft.class_3218$class_5526.method_31797(class_3218.java:1536) ~[server-intermediary.jar:?]
	at net.minecraft.class_5579.method_31850(class_5579.java:220) ~[server-intermediary.jar:?]
	at net.minecraft.class_5579$class_5580.method_31750(class_5579.java:115) ~[server-intermediary.jar:?]
	at net.minecraft.class_1297.method_31745(class_1297.java:3636) ~[server-intermediary.jar:?]
	at net.minecraft.class_1297.method_5650(class_1297.java:402) ~[server-intermediary.jar:?]
	at net.minecraft.class_1309.method_5650(class_1309.java:712) ~[server-intermediary.jar:?]
	at net.minecraft.class_1657.method_5650(class_1657.java:1408) ~[server-intermediary.jar:?]
	at net.minecraft.class_3218.method_18770(class_3218.java:929) ~[server-intermediary.jar:?]
	at net.minecraft.class_3324.method_14611(class_3324.java:413) ~[server-intermediary.jar:?]
	at net.minecraft.class_3244.method_10839(class_3244.java:1228) ~[server-intermediary.jar:?]
	at net.minecraft.class_2535.method_10768(class_2535.java:452) ~[server-intermediary.jar:?]
	at net.minecraft.class_3242.method_14357(class_3242.java:185) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:908) ~[server-intermediary.jar:?]
	at net.minecraft.class_3176.method_3813(class_3176.java:283) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:824) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:671) ~[server-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[server-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[08:05:31] [Server thread/ERROR]: This crash report has been saved to: /data/server/./crash-reports/crash-2023-07-10_08.05.31-server.txt
```

<html>
<body>

environment | version
-- | --
Java | azul/prime:17
HuskHomes | 4.4.1
Fabric | 0.14.21
Minecraft | 1.20.1

</body>
</html>

**Comment by foxii-cn** [comment]
The relevant configuration details are as follows:
```
  back_command_return_by_death: true
  back_command_save_teleport_event: false
  list_items_per_page: 12
  asynchronous_teleports: true
  play_sound_effects: true
```

**Comment by WiIIiam278** [comment]
I'm leaning towards this issue being caused more to do with proxy handling with fabric backends and less to do with HuskHomes. 

Homes isn't doing anything particularly _special_ -- it teleports players and uses standard plugin messages to request the proxy move the player between servers. None of the supporting logic to do that is actually in the plugin aside from just calling the relevant methods, so I wouldn't particularly know where to begin testing this.

**Comment by foxii-cn** [comment]
According to my observations, all current instances of this problem occur when there is cross-server and cross-dimensional teleportation at the same time: a player is at dimension X on sub-server A, uses the `/server` command to directly switch to sub-server B, then teleports to dimension Y on sub-server A, and this is when the problem occurs. Simply using `/server` to switch sub-servers does not cause this issue. This may potentially not be a problem with Huskhomes, but I wonder if you could help provide the main logic of how Huskhomes implements cross-server teleportation. I would like to see if I can build a simpler environment (for instance, composing a simple plugin that only commands certain sub-server and dimension teleportation) to reproduce this problem, then report it to the related Fabric project team.

**Comment by WiIIiam278** [comment]
Fixed in 4.4.2. Moved to using Fabric's dedicated API for doing cross-dimension teleports.
